digitalocean/droplets.go

```diff
@@ -0,0 +1,204 @@
+			Description: "state of the Droplet instance",
+		},
+		"locked": {
+			Type:        schema.TypeBool,
+			Description: "whether the Droplet has been locked",
+		},
+		"ipv4_address": {
+			Type:        schema.TypeString,
+			Description: "the Droplets public ipv4 address",
+		},
+		"ipv4_address_private": {
+			Type:        schema.TypeString,
+			Description: "the Droplets private ipv4 address",
+		},
+		"ipv6_address": {
+			Type:        schema.TypeString,
+			Description: "the Droplets public ipv6 address",
+		},
+		"ipv6_address_private": {
+			Type:        schema.TypeString,
+			Description: "the Droplets private ipv4 address",
+		},
+		"backups": {
+			Type:        schema.TypeBool,
+			Description: "whether the Droplet has backups enabled",
+		},
+		"ipv6": {
+			Type:        schema.TypeBool,
+			Description: "whether the Droplet has ipv6 enabled",
+		},
+		"private_networking": {
+			Type:        schema.TypeBool,
+			Description: "whether the Droplet has private networking enabled",
+		},
+		"monitoring": {
+			Type:        schema.TypeBool,
+			Description: "whether the Droplet has monitoring enabled",
+		},
+		"volume_ids": {
+			Type:        schema.TypeSet,
+			Elem:        &schema.Schema{Type: schema.TypeString},
+			Description: "list of volumes attached to the Droplet",
+		},
+		"tags": tagsDataSourceSchema(),
+		"vpc_uuid": {
+			Type:        schema.TypeString,
+			Description: "UUID of the VPC in which the Droplet is located",
+		},
+	}
+}
+
+func getDigitalOceanDroplets(meta interface{}) ([]interface{}, error) {
+	client := meta.(*CombinedConfig).godoClient()
+
+	opts := &godo.ListOptions{
+		Page:    1,
+		PerPage: 200,
+	}
+
+	var dropletList []interface{}
+
+	for {
+		droplets, resp, err := client.Droplets.List(context.Background(), opts)
+
+		if err != nil {
+			return nil, fmt.Errorf("Error retrieving droplets: %s", err)
+		}
+
+		for _, droplet := range droplets {
+			dropletList = append(dropletList, droplet)
+		}
+
+		if resp.Links == nil || resp.Links.IsLastPage() {
+			break
+		}
+
+		page, err := resp.Links.CurrentPage()
+		if err != nil {
+			return nil, fmt.Errorf("Error retrieving droplets: %s", err)
+		}
+
+		opts.Page = page + 1
+	}
+
+	return dropletList, nil
+}
+
+func flattenDigitalOceanDroplet(rawDroplet, meta interface{}) (map[string]interface{}, error) {
+	droplet := rawDroplet.(godo.Droplet)
+
+	flattenedDroplet := map[string]interface{}{
+		"name":          droplet.Name,
+		"urn":           droplet.URN(),
+		"region":        droplet.Region.Slug,
+		"size":          droplet.Size.Slug,
+		"price_hourly":  droplet.Size.PriceHourly,
+		"price_monthly": droplet.Size.PriceMonthly,
+		"disk":          droplet.Disk,
+		"vcpus":         droplet.Vcpus,
+		"memory":        droplet.Memory,
+		"status":        droplet.Status,
+		"locked":        droplet.Locked,
+		"created_at":    droplet.Created,
+		"vpc_uuid":      droplet.VPCUUID,
+	}
+
+	if droplet.Image.Slug == "" {
+		flattenedDroplet["image"] = droplet.Image.ID
```

Comment:
```suggestion
		flattenedDroplet["image"] = strconv.Itoa(droplet.Image.ID)
```


Working with both the droplet and droplets data sources, I'm running into:

```
Error: Unable to set `image` attribute: image: '' expected type 'string', got unconvertible type 'int'

  on main.tf line 8, in data "digitalocean_droplet" "drop":
   8: data "digitalocean_droplet" "drop" {
```